### DISCUSSION TITLE
Consider using conventional commits

### DISCUSSION THREAD

**Discussion by paulcsmith** [luckyframework/lucky#1115] [General] [answered]
https://www.conventionalcommits.org/en/v1.0.0-beta.4/

Generating changelogs and release notes can be time consuming and slows done our ability to release. Maybe we can use something the link above to keep things structured and auto-generate changelogs!

**Comment by stephendolan** [comment]
For the record, this is now out of beta!
https://www.conventionalcommits.org/en/v1.0.0/

**Comment by stephendolan** [comment]
@jwoertink This has been open for quite a while without many folks voicing disagreement... I'd be happy to open a PR with updated contribution guidelines, `script/test` checks, and GitHub Action lints if we want to give this a try.

Thoughts?

**Reply by jwoertink** [reply]
Yeah, I'll be honest... I won't (naturally) do this lol. It's a lot of extra thought in to writing a commit message when at work our team write most of our commits like "...". I try to be a bit more descriptive for Lucky's sake, but sometimes I'm just like "ugh, made a mistake.. my bad". Plus, I prefer to do the changelog myself. It doesn't really bother me to do it, and I don't always put every commit in to it. Commits that change internal things to Lucky's development are usually left out and the CHANGELOG just contains things that a user of Lucky would want to know. So things like "fixed typo in method docs" are sort of irrelevant and just add noise to the file. 

TL;DR - My vote would be to not do this; however, I'll let @paulcsmith make the final call on that. If he wants to do this, then I say we get started soon to get in the flow of it.

**Reply by paulcsmith** [reply]
Yeah let's close this. I agree on it being kind of a pain to do and unless we do it 100% of the time it won't really work 👍

**Reply by stephendolan** [reply]
Cool! Sorry for bringing some of these topics back from the dead :)

Trying to get one stale discussion/issue/PR moving per week to feel useful!

**Reply by jwoertink** [reply]
No worries! We appreciate it. Sometimes things just get set aside and forgotten about, so it's good to do a bit of cleanup here 😄

**Reply by paulcsmith** [reply]
Yeah cleanup is great. Thank you :)

**Comment by paulcsmith** [comment, marked as answer]
See https://github.com/luckyframework/lucky/discussions/1115#discussioncomment-82172
